File: src/web/src/main/java/io/zoemeow/pbl6/phonestoremanager/model/bean/ProductComment.java
package io.zoemeow.pbl6.phonestoremanager.model.bean;

import com.google.gson.annotations.SerializedName;

public class ProductComment {
    @SerializedName("id")
    Integer id;
    @SerializedName("userid")
    Integer userId;
    @SerializedName("productid")
    Integer productId;
    @SerializedName("comment")
    String comment;
    @SerializedName("rating")
    Float rating;
    @SerializedName("disabled")
    Boolean disabled;
    @SerializedName("disabledreason")
    String disabledReason;
    @SerializedName("user")
    User user;

    public Integer getId() {
        return id;
    }
    public void setId(Integer id) {
        this.id = id;
    }
    public Integer getUserId() {
        return userId;
    }
    public void setUserId(Integer userId) {
        this.userId = userId;
    }
    public Integer getProductId() {
        return productId;
    }
    public void setProductId(Integer productId) {
        this.productId = productId;
    }
    public String getComment() {
        return comment;
    }
    public void setComment(String comment) {
        this.comment = comment;
    }
    public Float getRating() {
        return rating;
    }
    public void setRating(Float rating) {
        this.rating = rating;
    }
    public Boolean getDisabled() {
        return disabled;
    }
    public void setDisabled(Boolean disabled) {
        this.disabled = disabled;
    }
    public String getDisabledReason() {
        return disabledReason;
    }
    public void setDisabledReason(String disabledReason) {
        this.disabledReason = disabledReason;
    }
    public User getUser() {
        return user;
    }
    public void setUser(User user) {
        this.user = user;
    }

    
}

File: src/api/Model/ProductComment.cs
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PhoneStoreManager.Model
{
    [Table("ProductComment")]
    public class ProductComment
    {
        [Key]
        [JsonProperty("id")]
        public int ID { get; set; }

        [Required]
        [JsonProperty("userid")]
        public int UserID { get; set; }

        [Required]
        [JsonProperty("productid")]
        public int ProductID { get; set; }

        [Required]
        [JsonProperty("comment")]
        public string Comment { get; set; } = string.Empty;

        [Required]
        [JsonProperty("rating")]
        public float? Rating { get; set; } = null;

        [JsonProperty("disabled")]
        public bool Disabled { get; set; } = false;

        [JsonProperty("disabledreason")]
        public string? DisabledReason { get; set; } = null;

        #region Entity
        [JsonProperty("user")]
        public User User { get; set; }

        [JsonIgnore]
        public Product Product { get; set; }
        #endregion
    }
}
